lib/iris/pandas.py

```diff
@@ -130,26 +130,18 @@ def _as_pandas_coord(coord):
 
 def _assert_shared(np_obj, pandas_obj):
     """Ensure the pandas object shares memory."""
-    if hasattr(pandas_obj, 'base'):
-        base = pandas_obj.base
+    if hasattr(pandas_obj, 'values'):
+        values = pandas_obj.values
     else:
-        base = pandas_obj[0].base
-
-    # Prior to Pandas 0.17, when pandas_obj is a Series, pandas_obj.values
-    # returns a view of the underlying array, and pandas_obj.base, which calls
-    # pandas_obj.values.base, returns the underlying array. In 0.17 and 0.18
-    # pandas_obj.values returns the underlying array, so base may be None even
-    # if the array is shared.
-    if base is None:
-        base = pandas_obj.values
+        values = pandas_obj[0].values
```

Comment:
Sorry @hdyson, but what is going on here? What are we getting back by getting the first item from the pandas object? What kind of object are you expecting this to be?

Reply:
A good question.  If we go back to the original implementation here:

    if hasattr(pandas_obj, 'base'):
        base = pandas_obj.base
    else:
        base = pandas_obj[0].base

then it's not clear to me what the if/else is protecting against either (bearing in mind that `.base` expands to `.values.base`, so the change in argument to hasattr from `base` to `values` has no affect on what pandas_obj refers to).  I assumed it had a use, though, so didn't want to discard it blindly.  

My interpretation is that `assert_shared` could be called with an argument of either a single pandas object (dataframe or series) or a sequence of them.  The only uses seem to be a single object though, so if you'd prefer for me to discard the if/else, and go to a simpler:

    base = pandas_obj.base

then I think we'll at least understand what the code is doing (and hopefully a test will fail if that breaks something).

Reply:
Your suggestion is appealing to be sure. I'm happy to go either way on this. On the whole, I'm pretty happy as is with the existing change. I'd like to get our Travis testing passing again before merging this, but otherwise 👍

Reply:
A little code archaeology, and it looks as though it was inherited from when pandas Series and Dataframes had a quite different data structure: https://github.com/SciTools/iris/commit/19a5487f7043bf6cdef29e3fb57671ec58e37905#diff-53a1661a36736be8a6017847069f7319R117 via a change to support numpy 1.7: https://github.com/SciTools/iris/commit/97ca7e9308b1b4fbf91ce5d9ddb3542615b011f3

I'll try killing it - my suspicion is that if it had some utility at one point, it probably doesn't any more.